### ISSUE TITLE
allow to generate l-trees in other materials except "air"

### ISSUE THREAD

**Issue by cpdef** [minetest/minetest#7190]
##### Issue type
 - Feature request
##### Minetest version
idk what i have to write here :)
i think the version doesn't matter,
but sorry its my first opened issue on minetest/minetest
##### Summary
Ok my Problem is as the title says, that you can not generate 
l-system trees inside any other material then "air", as paramat said to me.
U can generate the trunks there but only if the material is in the leaf group.

So for example underwater trees are not possible and 
If u have any gasslike material like i have on the marssurvive mod etc.
then u have a big problem generating there the tree.

**Comment by paramat** [comment]
Yes nodes are hardcoded to be placeable in 'air' or 'ignore' only.
Trunks are also placeable in 'leaf', 'leaf2' and 'fruit' nodes.
Placeable in 'airlike' nodes is certainly a good idea and is easy to code.

@cpdef i have a request, please could you replace 'u' with 'you' in the comment above?

**Comment by cpdef** [comment]
ok done
